src/Swift.Bindings/src/Parser/SwiftABIParser.cs

```diff
@@ -527,10 +523,13 @@ private static string ExtractUniqueName(string name)
             return name;
         }
 
-        private static string ExtractFullyQualifiedName(string parentName, string name)
-        {
-            return string.IsNullOrEmpty(parentName) ? ExtractUniqueName(name) : $"{parentName}.{ExtractUniqueName(name)}";
-        }
+        private static SwiftTypeName GetSwiftTypeName(BaseDecl parentDecl, string name)
+            => parentDecl switch
+            {
+                ModuleDecl moduleDecl => SwiftTypeName.FromModuleQualifiedName($"{moduleDecl.Name}.{name}"),
+                TypeDecl typeDecl => SwiftTypeName.FromModuleQualifiedName($"{typeDecl.SwiftTypeName.ModuleQualifiedName}.{name}"),
```

Comment:
Does `SwiftTypeName` exist at invocation time? Could we ensure it is not called if exists?

Reply:
I'm not sure I understand. you mean the parent's `SwiftTypeName` ?

Reply:
Sorry -- Is this only invoked at ABI parsing time? I wonder if it could be called multiple times.